arch/arm/core/cortex_m/mpu/arm_mpu_v7_internal.h

```diff
@@ -8,6 +8,11 @@
 #ifndef ZEPHYR_ARCH_ARM_CORE_CORTEX_M_MPU_ARM_MPU_V7_INTERNAL_H_
 #define ZEPHYR_ARCH_ARM_CORE_CORTEX_M_MPU_ARM_MPU_V7_INTERNAL_H_
 
+
+#define LOG_LEVEL CONFIG_ARCH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_DECLARE(mpu);
```

Comment:
Unfortunately, you cannot declare/register module in header file. It will end in conflict when you try to include such file in C file which declares logger module too.

This applies also to other subsystems.